### ISSUE TITLE
First argument to @ResolveProperty behaves unexpectedly

### ISSUE THREAD

**Issue by nurkkala** [nestjs/graphql#232]
## I'm submitting a...
<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[x] Documentation issue or request
[ ] Support request 
</code></pre>

## Current behavior

The first argument to `@ResolveProperty` appears to do nothing.

## Expected behavior

According to the documentation, the first argument should set the "real name" of the resolver.

> Conventionally, we would use something like getAuthor() or getPosts() as method names. We can easily do this by moving the real names to the decorators.

I assumed this meant that the name appearing first in the `@ResolveProperty` decorator would be used in the SDL output instead of the name of the resolver function itself. However, the only way I've found that works to override the resolver function's name is to use the `{ name: "..." }` option (last argument to `@ResolveProperty`).

What does the documentation mean by **real names**?

## Minimal reproduction of the problem with instructions

This results in SDL with `getCourses` as a property. The `courses` argument appears to do nothing.

```js
  @ResolveProperty("courses", () => [Course])
  async getCourses(@Parent() department: Department): Promise<Course[]> {
    return [new Course()]; //this.courseService.courses({ where: { department } });
  }
```

This results in an SDL property called `courses`, which is what I'm after.

```js
  @ResolveProperty(undefined, () => [Course], { name: "courses" })
  async getCourses(@Parent() department: Department): Promise<Course[]> {
    return [new Course()]; //this.courseService.courses({ where: { department } });
  }
```

## What is the motivation / use case for changing the behavior?

I'm not advocating for a change, just advocating for better understanding either through improved documentation or example code.

## Environment

<pre><code>
Nest version: 6.0.5
</code></pre>

**Comment by nurkkala** [comment]
Additional info: the following appears to work as advertised in the documentation:
```js
@ResolveProperty()
async courses(@Parent() department: Department) {
  //...
}
```
Also following the documentation, providing a "real name" for the resolver as follows:
```js
@ResolveProperty("courses")
async getCourses(@Parent() department: Department) {
  return [new Course()]; //this.courseService.courses({ where: { department } });
}
```
does *not* work, resulting in the message `Error: You need to provide explicit type for DepartmentResolver#getCourses`. Adding the suggested type:
```js
@ResolveProperty("courses", returns => [Course])
async getCourses(@Parent() department: Department) {
  return [new Course()]; //this.courseService.courses({ where: { department } });
}
```
resolves the error message, but now the SDL includes both `courses` and `getCourses` and the resolver function is named `getCourses`. Based on the documentation, I expect the resolver to be called `courses`. What am I missing?

**Comment by kamilmysliwiec** [comment]
It should be fixed in the @next version (`$ npm i @nestjs/graphql@next`). You will also have to update `@nestjs/core` to 6.2.0 (`$ npm i @nestjs/core@6.2.0`). Please, let me know if you face any issue.

**Comment by kamilmysliwiec** [comment]
Published as `6.2.0` (`$ npm i @nestjs/graphql@latest`). You will also have to update `@nestjs/core` to 6.2.0 (`$ npm i @nestjs/core@latest`)
